lib/animina/genservers/photo/photo.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Animina.GenServers.Photo do
+  @moduledoc """
+  This is the genserver that handles uploaded photos processing
+  """
+
+  use GenStage
+
+  require Ash.Query
+  alias Animina.Accounts
+  alias Animina.Accounts.Photo
+
+  @doc """
+  Starts a GenStage process linked to the current process.
+  """
+  def start_link do
+    GenStage.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  @doc """
+  Starts a GenStage process linked to the current process.
+  """
+  def start_link(_) do
+    GenStage.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  @doc """
+  Starts a queue which manages files to be processed
+  """
+  @impl true
+  def init(_) do
+    schedule_fetch_unprocessed_photos()
+
+    queue = :queue.new()
+    {:producer, {queue, 0}, dispatcher: GenStage.BroadcastDispatcher}
+  end
+
+  @impl true
+  def handle_call({:notify, photo}, from, {queue, pending_demand}) do
+    queue = :queue.in({from, photo}, queue)
+    dispatch_photos(queue, pending_demand, [])
+  end
+
+  @impl true
+  def handle_info({:fetch_unprocessed_photos, _from}, {queue, pending_demand}) do
+    schedule_fetch_unprocessed_photos()
+
+    bulk_result =
+      Photo
+      |> Ash.Query.for_read(:read)
+      |> Ash.Query.filter(state == ^:pending_review)
+      |> Accounts.read!(authorize?: false, page: [limit: 6])
```

Comment:
We should add an order by created_by here. The database otherwise will give any random 6 photos. But it should give us the 6 oldest photos.